apis/v1beta1/validation/gateway.go

```diff
@@ -91,3 +92,18 @@ func validateListenerHostname(listeners []gatewayv1b1.Listener, path *field.Path
 	}
 	return errs
 }
+
+// ValidateTLSCertificateRefs validates the certificateRefs
+// must be set and not empty when tls config is set and
+// TLSModeType is terminate
+func ValidateTLSCertificateRefs(listeners []gatewayv1b1.Listener, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	for i, c := range listeners {
+		if c.Protocol == gatewayv1b1.HTTPSProtocolType && c.TLS != nil {
```

Comment:
Why do you need to check the protocol type here?

Reply:
If I don't add it, it will still check certificatesRefs field when `TCP`, `HTTP`, `UDP` protocol. The outputs of unit tests:
```
FAIL    sigs.k8s.io/gateway-api/apis/v1alpha2/validation        0.013s
?       sigs.k8s.io/gateway-api/apis/v1beta1    [no test files]
ok      sigs.k8s.io/gateway-api/apis/v1beta1/util/translator    (cached)        coverage: 5.9% of statements
ok      sigs.k8s.io/gateway-api/apis/v1beta1/util/validation    0.031s  coverage: 100.0% of statements
--- FAIL: TestValidateGateway (0.00s)
    --- FAIL: TestValidateGateway/tls_config_present_with_http_protocol (0.00s)
        gateway_test.go:105: Expected 1 errors, got 2 errors: [spec.listeners[0].tls: Forbidden: should be empty for protocol HTTP spec.listeners[0].tls.certificateRefs: Forbidden: should be set and not empty when TLSModeType is Terminate
            ]
    --- FAIL: TestValidateGateway/tls_config_present_with_tcp_protocol (0.00s)
        gateway_test.go:105: Expected 1 errors, got 2 errors: [spec.listeners[0].tls: Forbidden: should be empty for protocol TCP spec.listeners[0].tls.certificateRefs: Forbidden: should be set and not empty when TLSModeType is Terminate
            ]
FAIL
```

Reply:
Then `TLS` and `GRPC` should be added to the check as well, no? You can use an approach similar to [this](https://github.com/kubernetes-sigs/gateway-api/blob/9e2805c66d603d2be9817e6209cb4bb0f2496562/apis/v1beta1/validation/gateway.go#L71) one to check all the supported protocols.